client/src/main/java/io/spine/client/SubscriptionObserver.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2019, TeamDev. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.client;
+
+import com.google.protobuf.Message;
+import io.grpc.stub.StreamObserver;
+
+import static io.spine.protobuf.AnyPacker.unpack;
+import static io.spine.util.Exceptions.unsupported;
+
+/**
+ * A {@link StreamObserver} of {@link SubscriptionUpdate} messages translating the message
+ * payload to the given delegate {@code StreamObserver}.
+ *
+ * <p>The errors and completion acknowledgements are translated directly to the delegate.
+ *
+ * <p>The {@linkplain SubscriptionUpdate#getEntityUpdates() messages} are unpacked
+ * and sent to the delegate observer one by one.
+ *
+ * @param <M>
+ *         the type of the delegate observer messages, which could be unpacked entity state
+ *         or {@code Event}
+ */
+final class SubscriptionObserver<M extends Message>
+        implements StreamObserver<SubscriptionUpdate> {
+
+    private final StreamObserver<M> delegate;
+
+    SubscriptionObserver(StreamObserver<M> targetObserver) {
+        this.delegate = targetObserver;
+    }
+
+    @SuppressWarnings("unchecked") // Logically correct.
+    @Override
+    public void onNext(SubscriptionUpdate value) {
+        SubscriptionUpdate.UpdateCase updateCase = value.getUpdateCase();
+        switch (updateCase) {
+            case ENTITY_UPDATES:
+                value.getEntityUpdates()
+                     .getUpdateList()
+                     .stream()
+                     .map(EntityStateUpdate::getState)
+                     .map(any -> (M) unpack(any))
```

Comment:
The `EntityStateUpdate` may not store a state but have `no_longer_matching` set to `true` instead. Looks like this line will fail in this case.

In general, it seems that we lack the way of telling the client that some entity state that previously matched his subscription criteria does not anymore.

Reply:
Well spotted! I've added the filtering condition, but this raises a question: “What do we do at the client, when no-longer-matching thing happen?”

Reply:
I'll file an issue for this because it's not a trivial thing, and I don't want to hold the client API because of this matter.

Reply:
Here's the issue on this matter: #1211